gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java

```diff
@@ -174,11 +175,9 @@ protected void configureExpectedParameters(FilterConfig filterConfig) {
     return audList;
   }
 
-  protected boolean tokenIsStillValid(JWT jwtToken) throws UnknownTokenException {
-    Date expires;
-    if (tokenStateService != null) {
-      expires = new Date(tokenStateService.getTokenExpiration(jwtToken));
-    } else {
+  protected boolean tokenIsStillValid(final JWT jwtToken) throws UnknownTokenException {
+    Date expires = getServerManagedStateExpiration(TokenUtils.getTokenId(jwtToken));
+    if (expires == null) {
```

Comment:
Is it possible here that an unknown token will return null and be considered still valid?

Reply:
I don't believe so; The UnknownTokenException should work its way back to the validateToken method where it is handled.